File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr1, vector<vector<int>> arr2) {
    vector<vector<int>> answer;
    
    answer.assign(arr1.size(), vector<int>(arr2[1].size(),0));
    
    int r = arr1.size();
    int c = arr2[0].size();
    int k = arr2.size();
    
    for(int i=0;i<r;i++){
        for(int t=0;t<c;t++){
            int sum=0;
            for(int j=0;j<k;j++){
                answer[i][t] += (arr1[i][j] * arr2[j][t]);
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.java
class Solution {
    public int[][] solution(int[][] arr1, int[][] arr2) {
        int[][] answer;
        
        int r1 = arr1.length;
        int c1 = arr1[0].length;
        
        int r2 = arr2.length;
        int c2 = arr2[0].length;
        
        answer = new int[r1][c2];
        
        for(int i=0;i<r1;i++){
            for(int j=0;j<c2;j++){     
                for(int t=0;t<c1;t++){
                    answer[i][j]+=(arr1[i][t] * arr2[t][j]);
                } 
            }
        }
        
        
        
        return answer;
    }
}